Backend/Middleware-APD/src/main/java/middlewareapd/util/ValidationUtil.java

```diff
@@ -0,0 +1,157 @@
+package middlewareapd.util;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.locks.ReadWriteLock;
+
+import com.auth0.jwt.interfaces.DecodedJWT;
+
+import middlewareapd.exception.UnauthorizedException;
+import middlewareapd.exception.UserNotFoundException;
+import middlewareapd.model.JWToken;
+import middlewareapd.repository.MockJWTRepository;
+
+/**
+ * ValidationUtil provides utility methods for validating JWT session data, UUIDs, and user roles.
+ * These methods are used to ensure that the JWT and its claims are consistent with the data stored in the database.
+ */
+public class ValidationUtil {
+
+    /**
+     * Validates the JWT session by checking if the JWT exists in the database
+     * and ensuring that the user has not logged out.
+     *
+     * @param retrievedJWT the JWT object retrieved from the database.
+     * @throws UnauthorizedException if the JWT is invalid, the session has expired, or the user is logged out.
+     */
+    public static void validateJwtSession(JWToken retrievedJWT) {
+        if (retrievedJWT == null) {
+            throw new UnauthorizedException("Invalid JWT or session expired.");
+        }
+
+        if (retrievedJWT.getLogout() != null) {
+            throw new UnauthorizedException("User has logged out.");
+        }
+    }
+
+    /**
+     * Validates the UUID by comparing the UUID extracted from the JWT with the one stored in the database.
+     *
+     * @param dbUuid the UUID stored in the database.
+     * @param extractedUuid the UUID extracted from the JWT.
+     * @throws UnauthorizedException if the UUIDs do not match.
+     */
+    public static void validateUuid(String dbUuid, String extractedUuid) {
+        if (!dbUuid.equals(extractedUuid)) {
+            throw new UnauthorizedException("JWT UUID mismatch.");
+        }
+    }
+
+    /**
+     * Validates the user's role by checking if the role from the database matches the role specified in the JWT.
+     *
+     * @param dbIsAdmin the admin status (role) stored in the database.
+     * @param isAdmin the admin status (role) extracted from the JWT.
+     * @throws UserNotFoundException if the user is not found for the provided UUID.
+     * @throws UnauthorizedException if the roles do not match.
+     */
+    public static void validateUserRole(Integer dbIsAdmin, int isAdmin) {
+        if (dbIsAdmin == null) {
+            throw new UserNotFoundException("User not found for the provided UUID.");
```

Comment:
sorry for the never read type behaviour, but where does this go to ah, like when u throw the exception

Reply:
Its gone ah but it shld throw to service then to controller to return error msg